### PR TITLE
CVQ2 141 specify pid in json

### PR DESCRIPTION
**Pull request by Rosa-Fox**
- DO NOT MERGE until [CVQ2 140](https://github.com/alphagov/verify-service-provider/pull/98) has been merged.

- Specify PID in JSON file for non-matching and as an argument for matching.

### CODE REVIEW

**Comment by alan-gds** [comment]
I'd encourage reviewers to have a think about whether we've taken the right approach in terms of where the PID (and other config) gets specified.

We don't want to PID to be specified in two places at once - that would be asking for trouble - so we don't take it as a parameter for the non-matching journey, as it's already required in the matching dataset json file.  Of course, we could have done the opposite and removed it from the matching dataset json file.  I think what we've done is more intuitive, since the json file now contains everything that is going to be returned in the test case responses (and nothing else besides).

It's worth keeping in mind that this script is only for our own convenience and is not likely to be used by RPs (although I presume they can see it if they're interested as the repo is public).  I think the current split between stuff specified in the script and stuff specified in the json file should be the most convenient one, but YMMV.

File: scripts/init-compliance-tool-for-environment.sh [hunk 1]
```diff
@@ -1,57 +1,88 @@
 #!/usr/bin/env bash
 set -e
 
-if [ ! $# -eq 2 ] ; then
-	echo "this script requires the environment and a persistentId to be passed in as arguments. e.g."
-	echo "$ ./init-compliance-tool.sh <environment> <persistentId>"
+if [ ! $# -eq 3 ] ; then
+	echo "For a matching journey this script must include the environment, the --matching flag and a PID"
+	echo "For example: $ ./init-compliance-tool-for-environment.sh <environment> --matching <persistentId>"
+	echo "For a matching journey this script must include the environment, the --nonmatching flag and the filepath to the matching dataset json"
```
Comment (RacingHippo) [inline scripts/init-compliance-tool-for-environment.sh]: Shouldn't that say "For a non-matching journey"?

Comment (alan-gds) [reply scripts/init-compliance-tool-for-environment.sh]: Good spot; now corrected.

**Comment by karlbaker02** [comment]
I agree that it's probably more intuitive that everything is in one place, although a counter argument could be made that to test different PIDs, initialising this via a command might be a quicker thing to change than the matching JSON file (in addition to giving you a history of initialisations of the compliance tool for varying PIDs, as opposed to changing the PID in a file, which has the effect of hiding the initialisation configurations).

In reality I think the setup of the PID in the JSON file is fine and is unlikely to cause any problems. Pending any changes that need to be made to PR #98, this looks good.

**Pull request merged**